src/OSPSuite.R/Domain/SimulationBatch.cs

```diff
@@ -69,12 +70,37 @@ ISimulationPersistableUpdater simulationPersistableUpdater
 
       public void Initialize(IModelCoreSimulation simulation, SimulationBatchOptions simulationBatchOptions)
       {
+         //SimModel optionally caches XML used for loading a simulation as string.
+         //This XML string was used e.g. by the old Matlab-/R-Toolbox when saving a simulation to XML.
+         //C++ export also depends on the original XML string at the moment (not quite clear why).
+         //Because per default XML is NOT cached, we need to set the KeepXML-option to true BEFORE loading a simulation.
+         _simModelBatch.KeepXMLNodeInSimModelSimulation = true;
+
          _simModelBatch.InitializeWith(simulation, simulationBatchOptions.Parameters, simulationBatchOptions.Molecules);
          //This needs to be done after initialization of the SimModelBatch so that we can check parameters
          validate(simulationBatchOptions);
          _simulationPersistableUpdater.UpdateSimulationPersistable(simulation);
       }
 
+      /// <summary>
+      /// Export model as C++ code; keep parameters and initial values set in InitializeWith as variable
+      /// </summary>
+      /// <param name="outputFolder">Model .cpp file will be created here</param>
+      /// <param name="fullMode">
+      ///   If true: all parameters will be set as to be varied before export (will only have effect if SimModel simulation was not finalized yet
+      ///   If false: parameters will be simplified (where possible)
+      /// </param>
+      /// <param name="modelName">
+      ///   If empty (default): model will be named Standard and exported to Standard.cpp
+      ///   Otherwise: model will be named to "modelName" in the C++ code and exported to modelName.cpp.
+      ///              modelName must be both valid file name AND valid C++ identifier in such a case
+      /// </param>
+      public void ExportToCPPCode(string outputFolder, bool fullMode, string modelName="")
+      {
+         var exportMode = fullMode ? CodeExportMode.Formula : CodeExportMode.Values;
+         _simModelBatch.ExportToCPPCode(outputFolder, exportMode, modelName);
+      }
```

Comment:
@PavelBal @msevestre **This** is the function which can be called from R to export the model (and variable parameters/initial values as they are set in the SimulationBatch) to C++.

With the current implementation of the SimulationBatch exporting in **Formula** and in **Value** mode would produce the same result I think (which corresponds to the Value mode) - which is ok anyway I think.

Exporting in **Formula** mode would make sense if e.g.
 - you would like to read the exported C++ code for whatever reason (I can hardly imagine any) or
 - if at the model creation time you don't know which parameters you will vary later and want to set ALL parameters as variable

Reply:
@Yuri05 Could you please explain what 
` valid C++ identifier in such a case` is (sorry if it's a trivial question).

>if at the model creation time you don't know which parameters you will vary later and want to set ALL parameters as variable

Setting all parameters as variable has a negative impact on the performance?

Reply:
>Could you please explain what

https://www.krellinst.org/archive/ais/textbook/unit7/CPP7.3.2.html

>Setting all parameters as variable has a negative impact on the performance?

Yes.